### PR TITLE
feat: Template merge functionality

### PR DESCRIPTION
**Pull request by rahulguptajss**

### CODE REVIEW

File: pkg/test/node_test.go [hunk 1]
```diff
@@ -45,3 +46,223 @@ func TestNode_Merge(t *testing.T) {
...
+	//export block
+
+	export := defaultTemplate.GetChildS("export")
+	if export != nil {
+		t.Errorf("missing export block")
+	}
+
+	if want9 != got9 {
+		t.Errorf("got %v, want %v", want9, got9)
+	}
```
Comment (Hardikl) [inline pkg/test/node_test.go:203]: which export is this?

Comment (rahulguptajss) [reply pkg/test/node_test.go:203]: saw it in some of zapiperf templates.

**Comment by rahulguptajss** [comment]
Refer https://github.com/NetApp/harvest-private/wiki/Template-Merge about the functionality

**Comment by rahulguptajss** [comment]
addressed review comments

**Review by cgrinds — COMMENTED** [review]
love to see all the tests. Nice job!

**Comment by rahulguptajss** [comment]
addressed review comments.

**Review by cgrinds — APPROVED** [review]
thanks @rahulguptajss 95% tests, 5% code \o/

**Comment by cgrinds** [comment]
docs need to be updated to reflect this change and how to use it
- how do i extend the default templates
- how do i replace one of the default templates

**Pull request merged**